3 - Assignment/240409 - Rust Basic + Roguelike Game, Week 2/expression-evaluation/src/lib.rs

```diff
@@ -22,7 +22,26 @@ enum Expression {
 }
 
 fn eval(e: Expression) -> Result<i64, String> {
-    todo!()
+    match e {
+    
+        Expression::Op { op, left, right } => {
+            match op {
+                Operation::Add => Result::Ok(eval(*left).expect("Error") + eval(*right).expect("Error")),
+                Operation::Sub => Result::Ok(eval(*left).expect("Error") - eval(*right).expect("Error")),
+                Operation::Mul => Result::Ok(eval(*left).expect("Error") * eval(*right).expect("Error")),
+                Operation::Div => {
+                    let rValue = eval(*right).expect("Error");
+                    if(rValue == 0){
+                        return Result::Err(String::from("Division by zero"))
+                    }
+                    Result::Ok(eval(*left).expect("Error") / rValue)
```

Comment:
1. Rust에서 변수는 `snake_case`를 사용합니다. 변수명을 변경해주세요.
2. `if`문에 `(`와 `)`는 없어도 됩니다. 없다고 오류가 나는 건 아니지만 Rust 스타일이 아니니 삭제해주세요.
3. `cargo fmt`를 통해 코드 스타일을 맞춰주세요.
4. `left`나 `right`의 결과가 `Ok`가 아니면 패닉이 발생하게 됩니다. 원하는 동작이 맞나요? 이 함수의 리턴 타입이 왜 `Result`인지 생각해보기 바랍니다.

```suggestion
                    let rvalue = eval(*right).expect("Error");
                    if rvalue == 0 {
                        return Result::Err(String::from("Division by zero"))
                    }

                    Result::Ok(eval(*left).expect("Error") / rvalue)
```